dashboard/app/models/script_level.rb

```diff
@@ -572,11 +565,9 @@ def summarize_for_teacher_panel(student, teacher = nil)
       id: level.id.to_s,
       contained: contained,
       submitLevel: level.properties['submittable'] == 'true',
-      paired: is_driver || is_navigator || false,
-      isDriver: is_driver,
-      isNavigator: is_navigator,
-      driver: driver,
-      navigators: navigators,
+      paired: !user_level.nil? && user_level.paired?,
+      partners: user_level&.partners,
+      partnerCount: user_level&.partner_count,
```

Comment:
It feels a little bit inaccurate to continue to call this pairing with partners now that there can be more than two in a group. I almost wish it was "collaborated" and "collaborators" to indicate groups of two or more, but totally understand if that's beyond the scope of this work.

Reply:
Or, are they only ever in a group of two, but maybe switch part way through a project e.g. Erin + James work together, then Erin + Maureen work together so partners = [James, Maureen]? Sorry for all the questions - I'm not totally caught up on what's been going on here 😄

Reply:
No worries at all, pairing has been (and continues to be) very confusing to me too!  :-)

The "paired" term comes from the curriculum where working in groups is called "paired programming".  The groups can have 2 or more students so we end up with "pairing groups" that have 2, or 3, or sometimes more students.  It is definitely confusing that something called a "pair" can have more than 2 people but I think using a different term might also be confusing.

Side note: In the code, there is also a strong distinction between the driver (the person who is logged in) and navigators.  It is intentional that this change blurs that line, moving us towards a model where everyone in a pairing group is an equal peer.

Reply:
Also, switching partners part way through a project is essentially not supported today.  Doing that will leave all the students with progress in a weird state.

Reply:
Thanks for clarifying. Yes, this makes sense now and agree consistency with how the groupings are talked about in the curriculum is the best route.